File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
	maxp := 0
	minb := prices[0]
	for i := 1; i < len(prices); i++ {
		minb = min(minb, prices[i])
		maxp = max(maxp, prices[i] - minb)
	}
	return maxp
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
    let i = 0, j = i + 1, len = prices.length, max = 0;
    while (j < len) {
        const diff = prices[j] - prices[i];
        max = Math.max(diff, max);
        if (diff < 0)
            i = j;
        j++;
    }
    return max;
};